ng-dev/pr/merge/pull-request.ts

```diff
@@ -73,22 +73,37 @@ export async function loadAndValidatePullRequest(
     return PullRequestFailure.claUnsigned();
   }
 
+  /** List of parsed commits for all of the commits in the pull request. */
+  const commitsInPr = prData.commits.nodes.map((n) => parseCommitMessage(n.commit.message));
+
+  const githubTargetBranch = prData.baseRefName;
   let targetLabel: TargetLabel;
-  try {
-    targetLabel = getTargetLabelFromPullRequest(config, labels);
-  } catch (error) {
-    if (error instanceof InvalidTargetLabelError) {
-      return new PullRequestFailure(error.failureMessage);
+  let targetBranches: string[];
+  if (!config.noTargetLabeling) {
```

Comment:
I'd probably swap the clauses here so that we do not double-negate.

Reply:
Also, how about extracting this into a dedicated function? That would also allow us to use that for testing, compared to manually having to handle the `noTargetLabeling` case in tests (this would give us more coverage).